arch/arm/src/stm32f0l0g0/stm32_adc.c

```diff
@@ -1132,6 +1155,19 @@ static void adc_common_cfg(struct stm32_dev_s *priv)
   uint32_t clrbits = 0;
   uint32_t setbits = 0;
 
+#if defined(CONFIG_STM32F0L0G0_STM32G0)
+
+  clrbits |= ADC_CCR_PRESC_MASK | ADC_CCR_VREFEN |
+             ADC_CCR_TSEN | ADC_CCR_VBATEN;
+
+#  ifdef STM32_ADC_CCR_PRESC
+  setbits |= STM32_ADC_CCR_PRESC;
+#  else
+  setbits |= ADC_CCR_PRESC_DIV4;
```

Comment:
why set some arbitrary prescaler value here? if user board needs prescaler then STM32_ADC_CCR_PRESC is for this. This whole section under `#if defined(CONFIG_STM32F0L0G0_STM32G0)` should be generalized and we shouldn't use chip specific ifdefs. You can use the same `clrbits` also for other chips